barretenberg/cpp/src/barretenberg/vm/avm/generated/flavor.cpp

```diff
@@ -2264,4 +2264,23 @@ AvmFlavor::PartiallyEvaluatedMultivariates::PartiallyEvaluatedMultivariates(cons
     }
 }
 
+AvmFlavor::ProvingKey::ProvingKey(const size_t circuit_size, const size_t num_public_inputs)
+    : circuit_size(circuit_size)
+    , evaluation_domain(bb::EvaluationDomain<FF>(circuit_size, circuit_size))
+    , commitment_key(std::make_shared<CommitmentKey>(circuit_size + 1))
+{
+    // These come from PrecomputedEntitiesBase, ugly that they are initialized like this!
```

Comment:
It might be ugly but the comment is not very helpful. Can you please be more specific about how it should be done?